290 update the node based on the core contracts changes

Vitomir2 opened the pull request:
# Description

- fetch the latest changes of the core-contracts, compile and run the tests;
- add the new functions and events that we will for the bindings;
- update the bindings and the initializers for the core contracts;
- update the commission command to accept new flags and to execute a set, apply, and claim commission;
- make the commission flag common, then add it in the register-validator command, and update the contract function parameters;
- optimize the functions encoding to directly use the contractsapi structs in order to easily set the input parameters;
- delete unused parameters;
- update the docs;

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

Vitomir2 committed:
adapt the node to the updated core-contracts

fetch the latest changes of the core-contracts, compile and run the tests;
add the new functions and events that we will for the bindings;
update the bindings and the initializers for the core contracts;
update the commission command to accept new flags and to execute a set, apply, and claim commission;
make the commission flag common, then add it in the register-validator command, and update the contract function parameters;
optimize the functions encoding to directly use the contractsapi structs in order to easily set the input parameters;
delete unused parameters;
update the docs;

Vitomir2 committed:
update git submodules commit

Vitomir2 committed:
delete redundant comment

Vitomir2 committed:
delete commented variable

R-Santev commented on README.md:
````diff
@@ -161,17 +161,21 @@ After your node is operational and fully synced, you're ready to become a valida
 ```
 
-The above command both register the validator and stakes the specified amount.
+The command above both registers the validator and stakes the specified amount. To set an initial commission rate that will be deducted from the future delegators' rewards, use the `--commission` flag with the desired value, as shown below:
````
add commission restrictions (from 0 to 100)

R-Santev commented on command/sidechain/commission/commission.go:
```diff
@@ -107,52 +123,88 @@ func runCommand(cmd *cobra.Command, _ []string) error {
 
-	receipt, err := setCommission(txRelayer, validatorAccount)
+	var input []byte
+	if params.commission > 0 { //nolint:gocritic
```
An user may want to set it to zero

R-Santev commented on command/sidechain/commission/params.go:
```diff
@@ -54,13 +60,30 @@ func (scp *setCommissionParams) validateFlags() error {
-	vals = append(vals, fmt.Sprintf("Staker Address|%s", scr.staker))
-	vals = append(vals, fmt.Sprintf("New Commission |%v", scr.newCommission))
+	if scr.isPending { //nolint:gocritic
+		buffer.WriteString("\n[NEW COMMISSION SET AND WILL REMAIN PENDING FOR 15 DAYS]\n")
```
Add important explanation that the user must execute apply tx after 15 days

R-Santev commented on command/sidechain/registration/register_validator.go:
```diff
@@ -70,6 +69,13 @@ func setFlags(cmd *cobra.Command) {
 		"stake represents amount which is going to be staked by the new validator account",
 	)
 
+	cmd.Flags().Uint64Var(
```
make the flag mandatory, so users don't forget to use it

Vitomir2 committed:
update

allow users to set 0 commission;
update output;
make the commission as a required flag for the register-validator command;

Vitomir2 committed:
update
